### PR TITLE
For #3481 - Implement swipe on toolbar to switch tabs.

### PR DESCRIPTION
**Pull request by person808**
Things to do
- [x] Swap swipe direction for rtl layout as described in the issue
- [x] Requires https://github.com/mozilla-mobile/android-components/pull/7534 for full-size thumbnail loading
- [x] Check if the tab preview toolbar needs to be a ConstraintLayout (double check rtl layout)
- [x] Fix thumbnail size when the thumbnails are taken from a different device orientation (ex. portrait thumbnails in landscape mode) (edit: reused the TabThumbnailView to scale the thumbnails)

Video - https://streamable.com/tk27vq

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by person808** [comment]
Updated with (hopefully) the final revision to my A-C PR. Tests will fail because the A-C changes break the tab tray. Will fix when we pick up the A-C changes.

File: app/src/main/java/org/mozilla/fenix/browser/TabPreview.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            )
+        )
+    }
+
+    override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
+        super.onLayout(changed, left, top, right, bottom)
+        previewThumbnail.translationY = if (!context.settings().shouldUseBottomToolbar) {
+            fakeToolbar.height.toFloat()
+        } else {
+            0f
```
Comment (boek) [inline app/src/main/java/org/mozilla/fenix/browser/TabPreview.kt:60]: linting might complain about this magic number

File: app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+    private val defaultVelocity = TypedValue.applyDimension(
+        TypedValue.COMPLEX_UNIT_DIP,
+        DP_PER_SECOND,
+        activity.resources.displayMetrics
+    )
+
+    private var gestureDirection = GestureDirection.LEFT_TO_RIGHT
+
+    init {
+        CoroutineScope(Main).launch {
```
Comment (boek) [inline app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt:85]: Are we launching here for a race condition? Is there another way we can fix it? (If not, we should probably at least attach it to a lifecycleScope)

File: app/src/main/java/org/mozilla/fenix/browser/TabPreview.kt (hunk 1 above)
Comment (person808) [reply app/src/main/java/org/mozilla/fenix/browser/TabPreview.kt:60]: Lint doesn't care it the magic number is 0 😄

File: app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt (hunk 2 above)
Comment (person808) [reply app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt:85]: Went with using a lifecycleScope. In theory the rootview should already be attached so we can get the insets right away, but I don't want to risk that not always being the case and crashing.

**Comment by person808** [comment]
Just an FYI for anyone on the perf team testing this. This requires building against a local AC that has https://github.com/mozilla-mobile/android-components/commit/800b465772f2faa0a48ecafd2e93c05a24e9d4ea until automation gets fixed and we pick up the next AC release

File: app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt (hunk 2 above)
Comment (person808) [reply app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt]: I found one scenario where the insets can change (switching from gesture -> three-button nav or vice-versa) so I decided to get the insets everytime I need to check them. To do that I changed `getInsets` so it'll just return null if the `decorView` isn't attached. In theory, we shouldn't run into this scenario and we'll be able to get the insets.

**Comment by andreicristianpetcu** [comment]
Can somebody please trigger a nightly build? Can't wait to test this!

**Pull request merged**